flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerTest.java

```diff
@@ -178,7 +179,7 @@ public void teardown() throws Exception {
 
 	static class TestingYarnResourceManager extends YarnResourceManager {
 		AMRMClientAsync<AMRMClient.ContainerRequest> mockResourceManagerClient;
-		NMClientAsync mockNMClient;
+		TestingNMClientAsync testingNMClientAsync;
```

Comment:
Good catch. It could be NMClientAsync in the current implementation.

Reply:
Since we construct this field in `TestingYarnResourceManager`, I think there is no need to keep it declared as `NMClientAsync` now. Besides, we need to access those `setter` functions of it now.